contracts/SportMarkets/SportsAMM.sol

```diff
@@ -564,7 +553,30 @@ contract SportsAMM is Initializable, ProxyOwned, PausableUpgradeable, ProxyReent
         uint additionalSlippage,
         address collateral
     ) public nonReentrant whenNotPaused {
-        _buyFromAMMWithDifferentCollateral(market, position, amount, expectedPayout, additionalSlippage, collateral);
+        _buyFromAMMWithDifferentCollateral(market, position, amount, expectedPayout, additionalSlippage, collateral, false);
+    }
+
+    /// @notice Buy amount of position for market/game from AMM using ETH
+    /// @param market The address of the SportPositional market of a game
+    /// @param position The position (home/away/draw) to buy from AMM
+    /// @param amount The position amount to buy from AMM
+    /// @param expectedPayout The amount expected to pay in sUSD for the amount of position. Obtained by buyAMMQuote.
+    /// @param additionalSlippage The slippage percentage for the payout
+    /// @param collateral The address of the collateral used
+    /// @param _referrer who referred the buyer to SportsAMM
+    function buyFromAMMWithEthAndReferrer(
+        address market,
+        ISportsAMM.Position position,
+        uint amount,
+        uint expectedPayout,
+        uint additionalSlippage,
+        address collateral,
+        address _referrer
+    ) external payable nonReentrant whenNotPaused {
+        if (_referrer != address(0)) {
+            IReferrals(referrals).setReferrer(_referrer, msg.sender);
+        }
+        _buyFromAMMWithDifferentCollateral(market, position, amount, expectedPayout, additionalSlippage, collateral, true);
```

Comment:
better to pass down `address(1)` or any hardcoded collateral, instead of using the collateral as input parameter

Reply:
The UI has to first fetch the quote using WETH collateral, so this is left for consistency.